libs/knowledge-store/ragstack_knowledge_store/directed_edge_extractor.py

```diff
@@ -101,7 +101,7 @@ def add_source_target_pairs(href_ids, url_ids):
                     for url_id in url_ids:
                         if not isinstance(url_id, str):
                             url_id = url_id.content_id
-                    source_target_pairs.add((href_id, url_id))
+                        source_target_pairs.add((href_id, url_id))
```

Comment:
I may try to look at the tests to see if I can handle this case -- my guess is it wasn't getting hit for some reason.